File: controller/models/custom.go
package models

import "time"

type WarehousePartWithName struct {
	WarehousePart
	PartName string `json:"part_name"`
}

type BikeWithModelName struct {
	Bike
	ModelName string `json:"model_name"`
}

type RoleManagementWithName struct {
	RoleManagement
	ProjectName string `json:"project_name"`
}

type OrderWithCustomer struct {
	Order
	CustomerName  string `json:"customer_name"`
	CustomerEmail string `json:"customer_email"`
}

type OrderItemsWithBikeName struct {
	OrderItem
	ModelName string `json:"model_name"`
}

type OrderItemsWithBikeAndDate struct {
	OrderItem
	ModelName string    `json:"model_name"`
	OrderDate time.Time `json:"order_date"`
}

type GraphMeta struct {
	CurrentRevenue  float64 `json:"current_revenue"`
	PreviousRevenue float64 `json:"previous_revenue"`
	CurrentSales    int     `json:"current_sales"`
	PreviousSales   int     `json:"previous_sales"`
}

type GraphData struct {
	Bucket  time.Time `json:"bucket"`
	Revenue float64   `json:"revenue"`
	SalesNo int       `json:"sales_no"`
}

type CityData struct {
	City            string  `json:"city"`
	CurrentRevenue  float64 `json:"current_revenue"`
	PreviousRevenue float64 `json:"previous_revenue"`
}

type BikeSales struct {
	BikeModel  string    `json:"bike_model"`
	OrderDate  time.Time `json:"order_date"`
	TotalSales int       `json:"total_sales"`
	Revenue    float64   `json:"revenue"`
}

File: frontend/src/types/custom.ts
import {Order, OrderItem} from "@/types/datatables";

export type OrderWithCustomer = Order & {
    customer_name: string;
    customer_email: string;
};

export type OrderItemsWithBikeName = OrderItem & {
    model_name: string;
};

export type OrderItemsWithBikeAndDate = OrderItem & {
    model_name: string;
    order_date: string;
};

export interface GraphMeta {
    current_revenue: number;
    previous_revenue: number;
    current_sales: number;
    previous_sales: number;
}

export interface GraphData {
    bucket: string;
    revenue: number;
    sales_no: number;
}

export interface CityData {
    city: string;
    current_revenue: number;
    previous_revenue: number;
}

export interface BikeSales {
    bike_model: string;
    order_date: string;
    total_sales: number;
    revenue: number;
}